lib/builders.js

```diff
@@ -79,18 +82,44 @@ class JSONBuilder {
 
   add(npmPackage) {
     const { name, version } = npmPackage;
-    const sources = npmPackage.licenseSources.license.sources;
 
-    const licenseText = formatLicenseText(sources.reduce((acc, source) => `${acc}${source.text}`, ''));
+    const licenseText = npmPackage.text ? formatLicenseText(npmPackage.text) : '';
 
     this.addCached(name, version, licenseText);
   }
 
+  buildLink(name, repository, licenseFile) {
+    return new Promise((resolve) => {
+      let licenseLink;
```

Comment:
You don't have to define this here. You define it at it's assignment on line 99 - could also use `const` then